src/classes/TDTM_ObjectDataGateway.cls

```diff
@@ -46,11 +46,11 @@ public with sharing class TDTM_ObjectDataGateway implements TDTM_iTableDataGatew
      * @return Boolean True is there are no TDTM records. False otherwise.
      */
     public Boolean isEmpty() {
```

Comment:
I wasn't able to find where this is used, since it uses a super common method name, but shouldn't it just pull the size of  listTH to avoid a new soql query?